src/base/bittorrent/peerinfo.cpp

```diff
@@ -60,17 +61,17 @@ PeerInfo::PeerInfo(const TorrentHandle *torrent, const libt::peer_info &nativeIn
 
 bool PeerInfo::fromDHT() const
 {
-    return (m_nativeInfo.source & libt::peer_info::dht);
+    return static_cast<bool>(m_nativeInfo.source & libt::peer_info::dht);
```

Comment:
I think `static_cast<>` is reasonable here. One thing to keep in mind is that an if-condition is a boolean context, which won't require the cast. So you could do this:

`return (m_nativeInfo.source & libt::peer_info::dht) ? true : false`

or just `return bool(m_nativeInfo.source & libt::peer_info::dht);`

Reply:
qBittorrent Coding style prohibits using function-style cast and implicit casting numbers to boolean.